src/main/java/org/jenkinsci/plugins/workflow/cps/CpsWhitelist.java

```diff
@@ -103,6 +104,9 @@ public boolean permitsStaticMethod(Method method, Object[] args) {
             String driverTo = "org.codehaus.groovy.runtime.";
             if (cn.startsWith(driverFrom)) {
                 try {
+                    if (Modifier.isPrivate(method.getModifiers())) {
+                        return true;
```

Comment:
Not sure why that would be necessary, though I have not yet studied it in much detail. I was thinking that [this](https://github.com/cloudbees/groovy-cps/blob/63b6aa1aa3b9414b4b2657ffbb5eeea39b21d115/dgm-builder/src/main/java/com/cloudbees/groovy/cps/tool/Translator.java#L151) would instead point to a subclass which creates a `FunctionCallEnv` with a `ProxyEnv` subtype overriding `getInvoker`.

Again, there would need to be an explicit test that, say,

```groovy
[1].each {x -> System.exit(666)}
```

is rejected.

BTW [this Javadoc](https://github.com/cloudbees/groovy-cps/blob/63b6aa1aa3b9414b4b2657ffbb5eeea39b21d115/lib/src/main/java/com/cloudbees/groovy/cps/impl/CpsCallable.java#L27) belongs in somebody’s hall of shame.